### ISSUE TITLE
[Accessibility] [Accordion] Controlled Accordion> Toggle state is not announced for 'Collapse All'/'Expand All' button

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#2038]
[A11Y-503](https://jira.cengage.com/browse/A11Y-503)

**Environment:**
Windows 11/Chrome 137/JAWS 2025
Windows 11/Edge 136/NVDA 2025
Mac 15/Safari/VoiceOver

**Steps to Reproduce:**

1. Open the URL – https://react-magma.cengage.com/version/4.9.1/api/tooltip/
2. Turn on the screen reader (JAWS/NVDA/VoiceOver).
3. Navigate to the ‘Accordion’ component page from the left navigation.
4. Locate the 'Collapse All' / 'Expand All' toggle button under ‘Controlled Accordion’.
5. Activate the button and observe the screen reader announcement.

**Actual Result:**
When 'Expand All' button is activated, screen reader does not automatically reads its toggle state i.e. 'Collapse All', rather it simply remains silent. Same is true when 'Collapse All' button is activated.

Refer to screencast:
https://app.screencast.com/m62RARqZ3Xx3c

**Expected Result:**

The screen reader should announce the toggle state of the 'Collapse All' / 'Expand All' button, indicating whether the accordions are currently expanded or collapsed.

User Impact:
Screen reader users are not informed about the outcome of activating the toggle, leading to confusion and poor navigation context regarding the expanded/collapsed state of content.

WCAG Guideline: 4.1.2 Name, Role, Value (A)

**Bug Fix Recommendation:**

a) Define the accessible name of 'Collapse All'/'Expand All' button using aria-label attribute.
b) Update this accessible name using javascript onClick of the button to announce the current state clearly.

**From Laura:**
Not a component issue, but we could update our examples to announce the change.

**Comment by olehnoskov-cengage** [comment]
**Tested on:**
- V4:
Docs: https://next--upbeat-sinoussi-f675aa.netlify.app/api/accordion/#controlled_accordion

- V5: 
Docs: https://rc--upbeat-sinoussi-f675aa.netlify.app/api/accordion/#controlled_accordion

Everything works as expected. No bugs found.

**Comment by silvalaura** [comment]
> We have reverified this issue on URL: [React Magma](https://next--upbeat-sinoussi-f675aa.netlify.app/) and observed that it is still not resolved. As per the current observation, role="status" is applied to the button, which overrides the native button semantics and prevents the screen reader from announcing it as a button for the Expand All component.
> 
> We recommend retaining the native <button> element without overriding its role. When the user selects the button and its label changes (e.g., Expand All / Collapse All), ensure that the updated button label is announced by the screen reader so that users are informed about the action performed.
> 
> Screencast: https://app.screencast.com/x1VVcpKESCJPW

**Comment by silvalaura** [comment]
> We have reverified this issue on URL: [React Magma](https://next--upbeat-sinoussi-f675aa.netlify.app/) and observed that it is not completely resolved. Currently, the role="status" is applied to the button, which overrides the native button semantics and prevents the screen reader from announcing it as a button. Additionally, the changes are now being announced by the screen reader.
> 
> Screencast: https://app.screencast.com/SEkmiwWg72OEG
